mypyc/ops_misc.py

```diff
@@ -176,85 +178,85 @@
           arg_types=[object_rprimitive, object_rprimitive],
           result_type=object_rprimitive,
           error_kind=ERR_MAGIC,
-          emit=simple_emit('{dest} = PyObject_GetItem({args[0]}, {args[1]});'),
+          emit=call_emit('PyObject_GetItem'),
           priority=0)
 
 method_op('__setitem__',
           arg_types=[object_rprimitive, object_rprimitive, object_rprimitive],
           result_type=bool_rprimitive,
           error_kind=ERR_FALSE,
-          emit=simple_emit('{dest} = PyObject_SetItem({args[0]}, {args[1]}, {args[2]}) >= 0;'),
+          emit=call_negative_bool_emit('PyObject_SetItem'),
           priority=0)
 
 method_op('__delitem__',
           arg_types=[object_rprimitive, object_rprimitive],
           result_type=bool_rprimitive,
           error_kind=ERR_FALSE,
-          emit=simple_emit('{dest} = PyObject_DelItem({args[0]}, {args[1]}) >= 0;'),
+          emit=call_negative_bool_emit('PyObject_DelItem'),
           priority=0)
 
 func_op(
     name='builtins.hash',
     arg_types=[object_rprimitive],
     result_type=int_rprimitive,
     error_kind=ERR_MAGIC,
-    emit=simple_emit('{dest} = CPyObject_Hash({args[0]});'))
+    emit=call_emit('CPyObject_Hash'))
 
 py_getattr_op = func_op(
     name='builtins.getattr',
     arg_types=[object_rprimitive, object_rprimitive],
     result_type=object_rprimitive,
     error_kind=ERR_MAGIC,
-    emit=simple_emit('{dest} = PyObject_GetAttr({args[0]}, {args[1]});')
+    emit=call_emit('PyObject_GetAttr')
 )
 
 py_setattr_op = func_op(
     name='builtins.setattr',
     arg_types=[object_rprimitive, object_rprimitive, object_rprimitive],
     result_type=bool_rprimitive,
     error_kind=ERR_FALSE,
-    emit=simple_emit('{dest} = PyObject_SetAttr({args[0]}, {args[1]}, {args[2]}) >= 0;')
+    emit=call_negative_bool_emit('PyObject_SetAttr')
 )
 
 func_op(
     name='builtins.hasattr',
     arg_types=[object_rprimitive, object_rprimitive],
     result_type=bool_rprimitive,
     error_kind=ERR_NEVER,
-    emit=simple_emit('{dest} = PyObject_HasAttr({args[0]}, {args[1]});')
+    emit=call_emit('PyObject_HasAttr')
 )
 
 py_delattr_op = func_op(
     name='builtins.delattr',
     arg_types=[object_rprimitive, object_rprimitive],
     result_type=bool_rprimitive,
     error_kind=ERR_FALSE,
-    emit=simple_emit('{dest} = PyObject_DelAttr({args[0]}, {args[1]}) >= 0;')
+    emit=call_negative_bool_emit('PyObject_DelAttr')
 )
 
 py_call_op = custom_op(
     arg_types=[object_rprimitive],
     result_type=object_rprimitive,
     is_var_arg=True,
```

Comment:
Somewhat unrelated: Is this correct? Should this always take 3 args?

Reply:
`PyObject_CallFunctionObjArgs` uses NULL-terminated varargs to pass an arbitrary number of positional parameters. `PyObject_Call` (below) always takes 3 args.